### PR TITLE
Database

### PR DESCRIPTION
**Pull request by Melvin-Viana**
If anyone is checking it I would appreciate some feedback about my models.

Or anything you'd like to point out?   :+1: :+1:

### CODE REVIEW

File: db/models/Product.js [hunk 1]
```diff
@@ -0,0 +1,29 @@
+module.exports = (sequelize, {
+  INTEGER, STRING, DATE, NOW,
+}) => {
+  const Product = sequelize.define('Product', {
```
Comment (AshbyGunter) [inline db/models/Product.js]: Is the product table supposed to represent some of the item data?

File: db/models/Review.js [hunk 2]
```diff
@@ -0,0 +1,23 @@
+module.exports = (sequelize, Sequelize) => {
+  const Review = sequelize.define('Review', {
```
Comment (AshbyGunter) [inline db/models/Review.js]: I'm not sure where the review scores are stored. That may be worth naming differently or making it it's own field.

File: db/index.js [hunk 3]
```diff
@@ -4,45 +4,18 @@ const { mySQLPassword, mySQLusername } = require('../config/index.js');
...
+const initializeDb = require('./initializeDb');
 
 const sequelize = new Sequelize(serviceDbName, mySQLusername, mySQLPassword, {
   dialect: 'mysql',
 });
-const Reviewer = require('./models/Reviewer')(sequelize, Sequelize);
-const Product = require('./models/Product')(sequelize, Sequelize);
-const ReviewPhoto = require('./models/ReviewPhotos')(sequelize, Sequelize);
-const ShopOwner = require('./models/ShopOwner')(sequelize, Sequelize);
-const Review = require('./models/Review')(sequelize, Sequelize);
```
Comment (AshbyGunter) [inline db/index.js]: good removal to another module - nice choice for separation of concerns

File: db/index.js [hunk 4]
```diff
@@ -4,45 +4,18 @@ const { mySQLPassword, mySQLusername } = require('../config/index.js');
...
 
-Review.hasOne(ReviewPhoto, { foreignKey: 'photo_id' });
-Review.hasOne(ShopOwner, { foreignKey: 'shopowner_id' });
-Review.hasOne(Product, { foreignKey: 'product_id' });
-Review.hasOne(Reviewer, { foreignKey: 'author_id' });
-ReviewPhoto.belongsToMany(Review, { through: 'ProductReviews' });
-ShopOwner.belongsToMany(Review, { through: 'ShopOwnerReviews' });
-ReviewPhoto.belongsTo(Review);
-Reviewer.belongsTo(Review);
-Product.hasOne(Product, { foreignKey: 'shopowner_id' });
```
Comment (AshbyGunter) [inline db/index.js]: good move, as above

File: db/initializeModels.js [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+  Review.hasOne(ShopOwner, { foreignKey: 'shopowner_id' });
+  Review.hasOne(Product, { foreignKey: 'product_id' });
+  Review.hasOne(Reviewer, { foreignKey: 'author_id' });
+  ReviewPhoto.belongsToMany(Review, { through: 'ProductReviews' });
+  ShopOwner.belongsToMany(Review, { through: 'ShopOwnerReviews' });
+  ReviewPhoto.belongsTo(Review);
+  Reviewer.belongsTo(Review);
+  Product.hasOne(Product, { foreignKey: 'shopowner_id' });
+  ShopOwner.belongsToMany(Product, { through: 'ShopOwnerProducts' });
+};
```
Comment (AshbyGunter) [inline db/initializeModels.js]: question - does all of this need to be exported, or is there a way to keep these definitions more contained?

File: db/initializeModels.js [hunk 6]
```diff
@@ -13,17 +13,17 @@ let Review = require('./models/Review');
...
+  Review.belongsTo(ReviewPhoto);
+  Review.belongsTo(Shop);
+  Review.belongsTo(Product);
+  Review.belongsTo(Reviewer);
+  ReviewPhoto.hasOne(Review, { foreignKey: 'photoId' });
+  Shop.hasMany(Review, { foreignKey: 'review_id' });
+  ReviewPhoto.hasOne(Review);
+  Reviewer.hasMany(Review);
+  Product.belongsTo(Shop);
+  Shop.hasMany(Product, { foreignKey: 'shop_id' });
```
Comment (AshbyGunter) [inline db/initializeModels.js]: much cleaner and clearer

File: db/initializeModels.js (hunk 5 above)
Comment (Melvin-Viana) [reply db/initializeModels.js]: Yes, this is a function that runs in my index.js. When it's run, all the Tables are created by Sequelize.

**Comment by Melvin-Viana** [comment]
Yeah thanks Ashby, So this seeding script adds the Database to the MySQL server if it doesn't exist. I made it so that it removes all data in the tables before any of the tables are populated.

Then after all that, my script populates all the tables with my helper functions in my utils folder.

The initializeModel.js file exports a function that is ran before I populate the database. It's responsible for all Schema stuff.

**Pull request merged**